### PR TITLE
Implement WebAuthn basic test using testdriver API

### PR DESCRIPTION
**Pull request by nsatragno**
This change adds a test for successfully creating a credential and
getting an assertion, automated using the testdriver.js WebAuthn API.

### CODE REVIEW

**Comment by nsatragno** [comment]
@jcjones please take a look.

What do you think about having `add_virtual_authenticator` not do anything if the browser running the test doesn't support the API? That way we could add code to automate the existing WPTs without breaking manual testing.

**Review by jcjones — APPROVED** [review]
Approved assuming the licensing is OK!

File: webauthn/resources/common-inputs.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    name: "madeline@example.com",
+    displayName: "Madeline",
+    icon: "https://example.com/celeste.png"
+};
+
+// ES256.
+const PUBLIC_KEY_PARAMETERS =  [{
+    type: "public-key",
+    alg: -7,
+},];
```
[the remark this answers is not in the document]
Comment (nsatragno) [reply webauthn/resources/common-inputs.js]: Done.

File: webauthn/resources/common-inputs.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    id: new TextEncoder().encode("123456789"),
+    name: "madeline@example.com",
+    displayName: "Madeline",
+    icon: "https://example.com/celeste.png"
+};
+
+// ES256.
+const PUBLIC_KEY_PARAMETERS =  [{
+    type: "public-key",
+    alg: -7,
```
[the remark this answers is not in the document]
Comment (nsatragno) [reply webauthn/resources/common-inputs.js]: Extracted into a constant on this file.

File: webauthn/resources/utils.js [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+           + (counter[1] << 16)
+           + (counter[2] << 8)
+           + counter[3],
+    attestedCredentialData,
+    extensions,
+  };
+}
+
+// Taken from
+// https://cs.chromium.org/chromium/src/chrome/browser/resources/cryptotoken/cbor.js?rcl=c9b6055cf9c158fb4119afd561a591f8fc95aefe
```
[the remark this answers is not in the document]
Comment (nsatragno) [reply webauthn/resources/utils.js]: I asked and it's fine :)

File: webauthn/webauthn-testdriver-basic.https.html [hunk 4]
```diff
@@ -0,0 +1,117 @@
+<!DOCTYPE html>
+<title>Successful WebAuthn tests</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script src="resources/common-inputs.js"></script>
+<script src="resources/utils.js"></script>
```
[the remark this answers is not in the document]
Comment (nsatragno) [reply webauthn/webauthn-testdriver-basic.https.html]: Yeah, it's intentional :( There's a `resources` folder in the `webauthn` folder with these scripts...

**Review by nsatragno — COMMENTED** [review]
Comments addressed:

**Comment by nsatragno** [comment]
FYI this is ready to merge @jcjones :smiley:  (I don't have write access to this repo)

**Review by bdstar81 — APPROVED** [review]

**Pull request merged**